### PR TITLE
Updates to VideoColorSpace attributes

### LINKED ISSUE #341: Make VideoColorSpace attributes non-nullable
Following https://github.com/w3c/webcodecs/pull/337, we realized that we currently have two ways to express "unknown color space": 1) as `null`, 2) as a VideoColorSpace with all `null` members. We should only have 1 way.

The second option is more coherent with other areas of the spec. 
a) when constructing a VideoColorSpace, you can pass {} or nothing at all to represent unknown. 
b) when constructing a VideoDecoderConfig, you can again pass {}.

A few other cleanups to do as part of this change
a) change VideoFrameBufferInit, VideoDecoderConfig, and VideoEncoderConfig IDL to default attribute colorSpace = {}. Update referencing algorithms accordingly.
b) take care to preserve VideoEncoder.encode()'s lazy assignment of colorSpace (take it from the first encode(frame) if config didnt have one)

### PR DESCRIPTION
**Pull request by chcunningham**
Makes colorSpace attributes non-nullables and removes
VideoEncoderConfig.colorSpace. See issues.

Fixes #341 and #345.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webcodecs/pull/346.html" title="Last updated on Aug 18, 2021, 6:23 PM UTC (5476d1a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webcodecs/346/fe0cb9c...5476d1a.html" title="Last updated on Aug 18, 2021, 6:23 PM UTC (5476d1a)">Diff</a>

### CODE REVIEW

File: index.src.html [hunk 1]
```diff
@@ -3454,7 +3428,8 @@
         |frame|.{{VideoFrame/timestamp}}.
     9. If |resource| has a known {{VideoColorSpace}}, assign its value to
         {{VideoFrame/[[color space]]}}.
-    10. Otherwise, assign `null` to {{VideoFrame/[[color space]]}}.
+    10. Otherwise, assign a new {{VideoColorSpace}}, contstructed with an emtpy
```
Comment (padenot) [inline index.src.html:3431]: `s/contstructed/constructed/`, `s/emtpy/empty/`

**Review by padenot — APPROVED** [review]
Approved modulo the two typos.

**Comment by chcunningham** [comment]
EC @aboba LGTM, will discuss possible follow up w/ internal color experts.

**Review by sandersdan — APPROVED** [review]

**Comment by chcunningham** [comment]
Thanks!

**Pull request merged**